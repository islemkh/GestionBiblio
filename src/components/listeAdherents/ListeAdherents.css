.adherentsTab-list{
margin-left: 100px;
width: 900px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  text-align: center;
}

.listeStyle{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    flex-wrap: wrap;
    
}