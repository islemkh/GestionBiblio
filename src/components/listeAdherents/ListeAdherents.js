import React from "react"
import Adherent from "../adherent/Adherent"
import './ListeAdherents.css'
 
function ListeAdherents(){
  var tabAdherents = localStorage.getItem("adherentsTab");
  var listaAdherents = JSON.parse(tabAdherents);
  return (
    <div className="adherentsTab-list">
      <h1>Listes des adherents</h1>
     <div className="listeStyle"  >
        {listaAdherents.map((adherentT, index)=> (
          <Adherent
            key={index}
              id={adherentT.id}
              nom={adherentT.nom}
              prenom={adherentT.prenom}
             statut={adherentT.statut}
             adheretsTab={listaAdherents}
            />
        ))}
      </div>
    </div>
  )

}
export default  ListeAdherents