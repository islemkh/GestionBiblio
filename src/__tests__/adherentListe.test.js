import React from 'react';
import { render } from '@testing-library/react';
import ListeAdherents from '../components/ListeAdherents/ListeAdherents';
import mockAdherent from '../mock/mockLivre'
import { createMemoryHistory } from "history"
import { Router } from 'react-router-dom';

describe("test list livres",()=>{
    test("test snapshots",()=>{
         const key ="adherentsTab"
        const adherent = [{"id":"1","nom":"Islem","prenom":"Khemiri","email":"islem@gmail.com","password":"123","statut":"active"},{"id":"2","nom":"Emna","prenom":"Amri","email":"emna@gmail.com","password":"123","statut":"active"},{"id":"3","nom":"Maryem","prenom":"Souyah","email":"maryem@gmail.com","password":"123","statut":"banni"},{"id":"4","nom":"Sana","prenom":"Kthiri","email":"sana@gmail.com","password":"123","statut":"active"},{"id":"5","nom":"ines","prenom":"ben mheni","email":"ines@gmail.com","password":"123","statut":"banni"},{"id":"6","nom":"Ahmed","prenom":"Amri","email":"ahmed@gmail.com","password":"123","statut":"active"}]
         localStorage.setItem(key, JSON.stringify(adherent)); 
          expect(localStorage.getItem("adherentsTab")).toBe(JSON.stringify(adherent))
       const history = createMemoryHistory({ initialEntries: ["/"] })
    const AdherentComponent = render(
      <Router history={history}> 
                <ListeAdherents Adherent={mockAdherent} />
      </Router>)              
        expect(AdherentComponent).toMatchSnapshot()    
  
    })    
  } )