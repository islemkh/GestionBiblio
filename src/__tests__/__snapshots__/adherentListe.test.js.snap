// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test list livres test snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="adherentsTab-list"
      >
        <h1>
          Listes des adherents
        </h1>
        <div
          class="listeStyle"
        >
          <div
            class="adherent"
          >
            <div>
              <img
                alt="profile"
                class="profilepic"
                src="im1.png"
              />
              <div
                class="title"
              >
                <p>
                  Islem
                   
                  Khemiri
                </p>
              </div>
            </div>
            <div
              class="action"
            >
              <button
                class="details"
              >
                details
              </button>
              <button
                class="banni"
              >
                bannir
              </button>
            </div>
          </div>
          <div
            class="adherent"
          >
            <div>
              <img
                alt="profile"
                class="profilepic"
                src="im1.png"
              />
              <div
                class="title"
              >
                <p>
                  Emna
                   
                  Amri
                </p>
              </div>
            </div>
            <div
              class="action"
            >
              <button
                class="details"
              >
                details
              </button>
              <button
                class="banni"
              >
                bannir
              </button>
            </div>
          </div>
          <div
            class="adherent"
          >
            <div>
              <img
                alt="profile"
                class="profilepic"
                src="im1.png"
              />
              <div
                class="title"
              >
                <p>
                  Maryem
                   
                  Souyah
                </p>
              </div>
            </div>
            <div
              class="action"
            >
              <button
                class="details"
              >
                details
              </button>
              <button
                class="debanni"
              >
                débannir
              </button>
            </div>
          </div>
          <div
            class="adherent"
          >
            <div>
              <img
                alt="profile"
                class="profilepic"
                src="im1.png"
              />
              <div
                class="title"
              >
                <p>
                  Sana
                   
                  Kthiri
                </p>
              </div>
            </div>
            <div
              class="action"
            >
              <button
                class="details"
              >
                details
              </button>
              <button
                class="banni"
              >
                bannir
              </button>
            </div>
          </div>
          <div
            class="adherent"
          >
            <div>
              <img
                alt="profile"
                class="profilepic"
                src="im1.png"
              />
              <div
                class="title"
              >
                <p>
                  ines
                   
                  ben mheni
                </p>
              </div>
            </div>
            <div
              class="action"
            >
              <button
                class="details"
              >
                details
              </button>
              <button
                class="debanni"
              >
                débannir
              </button>
            </div>
          </div>
          <div
            class="adherent"
          >
            <div>
              <img
                alt="profile"
                class="profilepic"
                src="im1.png"
              />
              <div
                class="title"
              >
                <p>
                  Ahmed
                   
                  Amri
                </p>
              </div>
            </div>
            <div
              class="action"
            >
              <button
                class="details"
              >
                details
              </button>
              <button
                class="banni"
              >
                bannir
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="adherentsTab-list"
    >
      <h1>
        Listes des adherents
      </h1>
      <div
        class="listeStyle"
      >
        <div
          class="adherent"
        >
          <div>
            <img
              alt="profile"
              class="profilepic"
              src="im1.png"
            />
            <div
              class="title"
            >
              <p>
                Islem
                 
                Khemiri
              </p>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              class="details"
            >
              details
            </button>
            <button
              class="banni"
            >
              bannir
            </button>
          </div>
        </div>
        <div
          class="adherent"
        >
          <div>
            <img
              alt="profile"
              class="profilepic"
              src="im1.png"
            />
            <div
              class="title"
            >
              <p>
                Emna
                 
                Amri
              </p>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              class="details"
            >
              details
            </button>
            <button
              class="banni"
            >
              bannir
            </button>
          </div>
        </div>
        <div
          class="adherent"
        >
          <div>
            <img
              alt="profile"
              class="profilepic"
              src="im1.png"
            />
            <div
              class="title"
            >
              <p>
                Maryem
                 
                Souyah
              </p>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              class="details"
            >
              details
            </button>
            <button
              class="debanni"
            >
              débannir
            </button>
          </div>
        </div>
        <div
          class="adherent"
        >
          <div>
            <img
              alt="profile"
              class="profilepic"
              src="im1.png"
            />
            <div
              class="title"
            >
              <p>
                Sana
                 
                Kthiri
              </p>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              class="details"
            >
              details
            </button>
            <button
              class="banni"
            >
              bannir
            </button>
          </div>
        </div>
        <div
          class="adherent"
        >
          <div>
            <img
              alt="profile"
              class="profilepic"
              src="im1.png"
            />
            <div
              class="title"
            >
              <p>
                ines
                 
                ben mheni
              </p>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              class="details"
            >
              details
            </button>
            <button
              class="debanni"
            >
              débannir
            </button>
          </div>
        </div>
        <div
          class="adherent"
        >
          <div>
            <img
              alt="profile"
              class="profilepic"
              src="im1.png"
            />
            <div
              class="title"
            >
              <p>
                Ahmed
                 
                Amri
              </p>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              class="details"
            >
              details
            </button>
            <button
              class="banni"
            >
              bannir
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
